server/api/src/rsDataObjOpen.cpp

```diff
@@ -391,6 +392,42 @@ namespace
                 }
             };
 
+            // Get the service role for the server to determine whether the catalog-generated information for
+            // the newly created replica is available in the L1 descriptor.
+            std::string service_role;
+            if (const auto ret = get_catalog_service_role(service_role); !ret.ok()) {
+                irods::log(PASS(ret));
+            }
+
+            // Need to get information from the catalog which was populated in the database operation if
+            // this server is not a catalog provider. In that instance, the dataObjInfo in the L1 descriptor
+            // could not have the information.
```

Comment:
just asking...  is it 'will not have' or 'cannot have' ?

Reply:
I think it's "will not have"